pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -418,16 +419,14 @@ public void sendComplete(Exception e) {
                         stats.incrementNumAcksReceived(latencyNanos);
                     }
                 } finally {
-                    interceptorMessage.getDataBuffer().release();
+                    ReferenceCountUtil.safeRelease(payloadInCurrentMsg);
```

Comment:
I think the correct fix should be always release the buffer before completing the future.
It doesn't make sense to allow users to release the message first, and then we try to touch the message again.

Reply:
Maybe we can calculate the metric once time for one batch message.

Reply:
And fixed the inaccurate metrics: latencyNanos

Reply:
> Maybe we can calculate the metric once time for one batch.

Good suggestion. I think calculating for every single message is better, the metrics `publish latency` will be more precise when users set `batchingMaxPublishDelay` to a large value.

Reply:
Yes.

Just a thought: if the publish latency contains the batch delay, the metric can't accurately reflect the latency of communication with the broker, but maybe it's OK because it also can reflect the trend.